consensus/core/src/commit.rs

```diff
@@ -204,6 +204,22 @@ impl LeaderStatus {
             Self::Undecided(_) => false,
         }
     }
+
+    pub fn into_decided_slot(self) -> Slot {
```

Comment:
This should be a getter instead.

Reply:
Getter sounds good. An alternative I can propose is implementing the `From<LeaderStatus> for Slot` . Then you can easily call `leader_status.into()` whenever a `Slot` is required

Reply:
Personally I dislike implementing `From` for very different types. A simple function or getter may be clearer in a lot of the cases.

Reply:
I think you guys are just referring to changing the name of the method and not the semantics itself. Changed to `get_decided_slot` & `get_committed_block`

Reply:
Clarified with Mingwei offline and properly changed `get_decided_slot`  to a getter and reverted `into_committed_block`